### PR TITLE
some sequence diagram to capture current web app interactions

### PR DESCRIPTION
**Pull request by sgandon**
**What is the new behavior?**
new sequence diagrams to analyse the serialization issues of component properties in web apps and issues with enrichment processes of middleware on top of tcomp ui-properties.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [x] Other... Please describe:
new documentation added

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/798/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/798/)

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Creating a Dataset instance
+=====================
+
+here we can see how Data Stream and Data Prep are using tcomp for creating a dataset. This document will focus on the tcomp interaction and the serialization/storage process done by the middleware to show that today's approach is costly and hard to maintain.
```
Comment (gonchar-ivan) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: Please add here after middleware (Data Stream and Data Prep back)

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+* **[004]** Get the UI specs for that dataset from TCOMP. 
+* **[008]** The user modify them.
+* **[011]** The dataset *ui-properties* are persisted.
+
+The issues with this storage approach is that the Middleware do not have knowledge of the field types and for example do not know which one should be encrypted before storage for security reasons but tcomp knows.
+Another issue is the Migration path, not enought information is stored in the *ui-properties* in order to let tcomp migrate the data.
+
+This leads to the conclusion that the data format used to store the properties by the middleware application is not the right one. 
+ 
+![Overwiew UI sequence: Create a Dataset](dataset_create_overwiew_2017_06.png)
```
Comment (gonchar-ivan) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: not a big trouble, but a type on the picture:
[004] POST /properties/JDBCDatastore/dataset?**form**Name

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+### 2) The proposed solution for a proper storage sequence
+in  blue the payload is a serialized version of the properties that is ecrypted, can be migrated and is compatible with the Studio persistence.
+Summary
+* **[003]** retrieve datastore properties from persistent storage in the form of a *tcomp-ser-properties*.
+* **[004]** Get the UI specs for that dataset from TCOMP.
```
Comment (gonchar-ivan) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: Will be used only by DataPrep/DataStream?

Is it possible that some client will require to push plain properties (not encrypted)

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+
+### 2) The proposed solution for a proper storage sequence
+in  blue the payload is a serialized version of the properties that is ecrypted, can be migrated and is compatible with the Studio persistence.
+Summary
+* **[003]** retrieve datastore properties from persistent storage in the form of a *tcomp-ser-properties*.
+* **[004]** Get the UI specs for that dataset from TCOMP. 
+* **[008]** The user modify them.
+* **[011] get the serialized version of the *ui-properties* from TCOMP.**
+* **[013]** Store the *tcom-ser-properties*.
```
Comment (gonchar-ivan) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: typo: tcomp

*tcomp-ser-properties* are properties in json-io format, right?

File: docs/specs/front-tcomp/dataset-edit-with-detailed-payload.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+Modifying a Dataset instance
```
Comment (gonchar-ivan) [inline docs/specs/front-tcomp/dataset-edit-with-detailed-payload.md]: This part seems ok for me, but I didn't get proposal described in https://jira.talendforge.org/browse/TDKN-168

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md (hunk 3 above)
Comment (sgandon) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: has mentionned in the slack channel #comp-service we suggest that the properties gets encrypted on the backend side where they get stored.

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md (hunk 4 above)
Comment (sgandon) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: yes they are

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/909/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/909/)

File: docs/specs/front-tcomp/dataset-create-overview-proposal.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+### 2) The proposed solution for a proper storage sequence
+in  blue the payload is a serialized version of the properties that is ecrypted, can be migrated and is compatible with the Studio persistence.
+Summary
+* **[003]** retrieve datastore properties from persistent storage in the form of a *tcomp-ser-properties*.
+* **[004]** Get the UI specs for that dataset from TCOMP. 
+* **[008]** The user modify them.
+* **[011] get the serialized version of the *ui-properties* from TCOMP.**
+* **[013]** Store the *tcomp-ser-properties*.
+
+here the is an extra call from the middleware to tcomp in order to have a proper serializable payload  that will have encrypted field whenever required, a migration path already implemented and a compatiblity with studio persistance.
```
Comment (gonchar-ivan) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: "that will have encrypted field whenever required"
As I understand with new proposal in tcomp-service this [011] endpoint won't encrypt anything.
Am I right?

Comment (sgandon) [inline docs/specs/front-tcomp/dataset-create-overview-proposal.md]: yes you a right, I'll update this with this possible solution too.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/910/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/910/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/913/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/913/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/914/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/914/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/915/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/915/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/916/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/916/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/917/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/917/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/918/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/918/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/919/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/919/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/920/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/920/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/921/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/921/)

**Pull request merged**